src/main/java/com/facebook/openwifirrm/ucentral/UCentralUtils.java

```diff
@@ -79,10 +83,45 @@ public static List<WifiScanEntry> parseWifiScanEntries(
 			for (JsonElement e : scanInfo) {
 				WifiScanEntry entry = gson.fromJson(e, WifiScanEntry.class);
 				entry.unixTimeMs = timestampMs;
+				// extract relevant IEs
+				JsonArray ies = e.getAsJsonObject().get("ies").getAsJsonArray();
+				for (JsonElement ie : ies) {
+					JsonElement typeElement = ie.getAsJsonObject().get("type");
+					if (typeElement == null) {
+						continue;
+					}
+					final String contentFieldKey = "content";
+					switch (typeElement.getAsInt()) {
+					case Country.TYPE:
+						entry.country = gson.fromJson(
+							ie.getAsJsonObject().get(contentFieldKey),
+							Country.class
+						);
+						break;
+					case QbssLoad.TYPE:
+						entry.qbssLoad = gson.fromJson(
+							ie.getAsJsonObject().get(contentFieldKey).getAsJsonObject().get("802.11e CCA Version"),
```

Comment:
can you move the parsing of the inner "802.11e CCA Version" part into the `QbssLoad` class as well? otherwise it feels like you're leaking implementation details into this parser function